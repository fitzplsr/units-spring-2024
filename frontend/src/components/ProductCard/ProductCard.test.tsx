import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { ProductCard } from './ProductCard';
import { Product } from '../../types';
import { getPrice } from '../../utils';

jest.mock('../../utils/getPrice', () => ({
    __esModule: true,
    getPrice: jest.fn(() => '999 $'),
}));

const product: Product = {
    id: 1,
    name: 'IPhone 14 Pro',
    description: 'Latest iphone, buy it now',
    price: 999,
    priceSymbol: '$',
    category: 'Электроника',
    imgUrl: '/iphone.png',
};

afterEach(jest.clearAllMocks);
describe('Product card test', () => {
    it('should render correctly', () => {
        const rendered = render(<ProductCard key={product.id} {...product} />);
        expect(rendered.asFragment()).toMatchSnapshot();
    });

    it('should call callback when render', () => {
        render(<ProductCard key={product.id} {...product} />);
        expect(getPrice).toHaveBeenCalledTimes(1);
    });
});
