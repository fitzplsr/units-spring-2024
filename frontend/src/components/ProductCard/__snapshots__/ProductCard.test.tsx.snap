// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product card test should render correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
  >
    <div
      class="product-card__text"
    >
      <h2
        class="product-card__name"
      >
        IPhone 14 Pro
      </h2>
      <div
        class="product-card__description"
      >
        Latest iphone, buy it now
      </div>
      <h1
        class="product-card__price"
      >
        999 $
      </h1>
      <div
        class="product-card__category"
      >
        Электроника
      </div>
    </div>
    <img
      alt="IPhone 14 Pro"
      class="product-card__image"
      src="/iphone.png"
    />
  </div>
</DocumentFragment>
`;
